contributors/devel/strategic-merge-patch.md

```diff
@@ -209,15 +211,115 @@ In an erroneous case, the set may be created with duplicates. Deleting an
 item that has duplicates will delete all matching items.
 
 
-TODO: @pwittrock
-# Changing Patch Format
+# Changing patch format
 
-## Purpose
+As issues and limitations have been discovered with the strategic merge
+patch implementation, it has been necessary to change the patch format
+to support additional semantics - such as merging lists of
+primitives and defining order when merging lists.
 
-## Requirement
+## Requirments for any changes to the patch format
 
-### Version Skew
+**Note:** Changes to the strategic merge patch must be backwards compatible such
+that patch requests valid in previous versions continue to be valid.
 
-## Strategy
+Previously valid patch requests do not need to keep the exact same
+behavior, but do need to behave correctly.
+
+**Example:** if a patch request previously randomized the order of elements
+in a list and we want to provide a deterministic order, we must continue
+to support old patch format but we can make the ordering deterministic
+for the old format.
+
+### Client version skew
+
+Because the server does not publish which patch versions it supports,
+and it silently ignores patch directives that it does not recognize,
+new patches should behave correctly when sent to old servers that
+may not support all of the patch directives.
+
+`kubectl` is a client that must support talking to older and newer server versions,
+and generate patches that work correctly on both.
+
+## Strategies for introducing new patch behavior
+
+#### 1. Add optional semantic meaning to the existing patch format.
+
+**Note:** Must not require new data or elements to be present that was not required before.  Meaning must not break old interpretation of old patches.
+
+**Good Example:**
+
+Old format
+  - ordering of elements in patch had no meaning and the final ordering was arbitrary
+
+New format
+  - ordering of elements in patch has meaning and the final ordering is deterministic based on the ordering in the patch
+
+**Bad Example:**
+
+Old format
+  - fields not present in a patch for Kind foo are ignored
+  - unmodified fields for Kind foo are optional in patch request
+
+New format
+  - fields not present in a patch for Kind foo are cleared
+  - unmodified fields for Kind foo are required in patch request
+
+This example won't work, because old patch formats will contain data that is now
+considered required.  To support this, introduce a new directive to guard the
+new patch format.
+
+#### 2. Add support for new directives in the patch format
+
+- Optional directives may be introduced to change how the patch is applied by the server - **backwards compatible** (old patch against newer server).
+  - *Old servers will ignore directives they do not recognize*
+  - May control how the patch is applied
+  - May contain optional metadata such as a list of elements to delete from a list that was not present before
```

Comment:
>that was not present before

I feel this clause makes this sentence not very clear. May we should rewording it.